src/extensions/extension-discovery.ts

```diff
@@ -364,13 +364,12 @@ export class ExtensionDiscovery extends Singleton {
       const absolutePath = (isProduction && await fse.pathExists(npmPackage)) ? npmPackage : extensionDir;
       let isCompatible = isBundled;
 
-      if (manifest.engines?.lens) {
+      /* assume bundled extensions are compatibile */
+      if (!isBundled && manifest.engines?.lens) {
         const appSemVerLatestImplied = appSemVer;
```

Comment:
Does this actually copy it, if not why not just clear it directly?

Reply:
We probably don't want to mutate `appSemVer` and that's why it's copied.

Reply:
But this is JS all we are doing here is copying the object pointer.

example:

```js
const x = {
	foo: [1, 2, 3]
};

console.log(x);

const y = x;

y.foo = [];

console.log(y);
console.log(x);
console.log(y === x);
```

Reply:
pointer? :)

Yes, javascript trips me up again. I will rework...

Reply:
Yup pointer, all object references are pointers which are always passed by value

Reply:
If it's a pointer I should be able to do this:
```
y->foo = [];
```

I'm just kidding, you're talking about pointers in javascript. (I think I know what a reference is :) )